clients/da-vinci-client/src/main/java/com/linkedin/davinci/kafka/consumer/ActiveActiveStoreIngestionTask.java

```diff
@@ -1283,4 +1268,52 @@ Runnable buildRepairTask(
           upstreamOffset);
     };
   }
+
+  int getRmdProtocolVersionID() {
+    return rmdProtocolVersionID;
+  }
+
+  protected ProduceToTopic getProduceToTopicFunction(
+      byte[] key,
+      ByteBuffer updatedValueBytes,
+      ByteBuffer updatedRmdBytes,
+      int valueSchemaId,
+      boolean resultReuseInput) {
+    return (callback, sourceTopicOffset) -> {
+      final ChunkAwareCallback newCallback = new ChunkAwareCallback() {
+        @Override
+        public void setChunkingInfo(
+            byte[] key,
+            ByteBuffer[] valueChunks,
+            ChunkedValueManifest chunkedValueManifest,
+            ByteBuffer[] rmdChunks,
+            ChunkedValueManifest chunkedRmdManifest) {
+          ((ChunkAwareCallback) callback)
```

Comment:
Can we update the definition of `ProduceToTopic` interface that the first callback input must be an instance of `ChunkAwareCallback`? Then we don't need this casting.